slatedb/src/utils.rs

```diff
@@ -98,7 +98,11 @@ where
 }
 
 pub(crate) fn system_time_to_millis(system_time: SystemTime) -> i64 {
-    system_time.duration_since(UNIX_EPOCH).unwrap().as_millis() as i64
+    if system_time < UNIX_EPOCH {
+        -(UNIX_EPOCH.duration_since(system_time).unwrap().as_millis() as i64)
```

Comment:
what is this about?

Reply:
Haha, ya, sorry this is what I meant to leave a comment on. The issue I found was that `system_time_to_millis` didn't work for times less than the Unix epoch. We've already accommodated this problem in the DefaultSystemClock::new. You basically need to swap unix epoch and system_time and then negate to make pre-unix epoch timestamps work. I didn't bother adding a test because the unit tests in the mock verify it already.

Here's how it's handled in new():

```
        let ts_millis = match SystemTime::now().duration_since(UNIX_EPOCH) {
            Ok(duration) => duration.as_millis() as i64, // Time is after the epoch
            Err(e) => -(e.duration().as_millis() as i64), // Time is before the epoch, return negative
        };
```

Reply:
Fixed: eab3077e9a71772ad8b209d39de3f052d46f008b